src/Tokenizer/BaseTokenizer.cs

```diff
@@ -296,6 +286,61 @@ public List<long> ConvertTokensToIds(List<string> tokens)
         };
     }
 
+    /// <summary>
+    /// Builds input with special tokens.
+    /// </summary>
+    public TokenIdsWithSpecialTokens BuildInputWithSpecialTokens(TokenIdsWithOffsets tokens1, TokenIdsWithOffsets tokens2 = null)
```

Comment:
We want to remove the `XLMRobertaTokenizer tokenizer` parameter from `encode` but we want to keep the possibility to implement another tokenizer in the future. A tokenizer can have a custom method `BuildInputWithSpecialTokens` as it is the case for the `XLMRobertaTokenizer`.
I would recommend having a virtual `BuildInputWithSpecialTokens` in `BaseTokenizer` [with this implementation](https://github.com/guillaume-be/rust-tokenizers/blob/main/main/src/tokenizer/base_tokenizer.rs#L1168) and keeping this one in `XLMRobertaTokenizer`  and adding the `override` keyword to the method, just like for `TokenizeToTokens`.

Reply:
Agree 100%
Done in next commit.